iothub/device/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -51,6 +50,13 @@ public bool Equals(SharedAccessSignatureAuthorizationRule other)
             return equals;
         }
 
+        public override bool Equals(object obj)
+        {
+            return obj is SharedAccessSignatureAuthorizationRule
```

Comment:
if obj is of type SharedAccessSignatureAuthorizationRule, wouldn't the other 'Equals' method be called instead:
`public bool Equals(SharedAccessSignatureAuthorizationRule other)`

If so, that means this Equals will always throw because it will always be called when the passed in object is not SharedAccessSignatureAuthorizationRule.

I could be way off here and my cold symptoms are speaking :D

Reply:
I guess it depends on the caller calls this method. Whether he passes in an object type (but underneath is of type SharedAccessSignatureAuthorizationRule), or passes in a SharedAccessSignatureAuthorizationRule type.

if that's the case, should line 56 be `Equals(obj as SharedAccessSignatureAuthorizationRule)`?

Reply:
You are not wrong, the only caviat is that if you pass in an instance of the SharedAccessSignatureAuthorizationRule as an object, it will hi that path. 
The main reason we now have this method override is to basically get rid of the warning

Reply:
I opted to not throw the exception and always call the implemented model with the SharedAccessSignatureAuthorizationRule as an input.